tests/lib/prepare-restore.sh

```diff
@@ -565,6 +563,145 @@ restore_suite_each() {
         fi
         sleep 1
     done
+
+    # shellcheck source=tests/lib/reset.sh
+    "$TESTSLIB"/reset.sh --reuse-core
+
+    # Reset failing orphan services before checks
+    systemctl reset-failed
+
+    do_checks
+}
+
+get_logs_file() {
+    logs_dir="$RUNTIME_STATE_PATH/logs"
+    logs_id=$(find "$logs_dir" -maxdepth 1 -name '*.journal.log' | wc -l)
+    logs_file=$(echo "${logs_id}_${SPREAD_JOB}" | tr '/' '_' | tr ':' '__')
+
+    echo "${logs_dir}/${logs_file}"
+}
+
+save_pkgs() {
+    file=$1
```

Comment:
Shellcheck would probably complain about missing quotes around $1, i.e. I think this should be `file="$1"`. Or am I missing something? (here and in some more places).